GVFS/GVFS.Service/GvfsService.cs

```diff
@@ -451,7 +451,7 @@ private void MigrateUpgradeLogsToDirectoryWithFreshACLs()
                 else
                 {
                     PhysicalFileSystem fileSystem = new PhysicalFileSystem();
-                    fileSystem.DeleteDirectory(tempUpgradeLogsPath, recursive: false);
+                    fileSystem.RecursiveDelete(tempUpgradeLogsPath);

```

Comment:
This is actually one spot that I don't think we want `RecursiveDelete` because we want the delete to fail if there are any files left in the old Logs directory (it's currently a possible race condition).

Rather than removing `DeleteDirectory`, what would you think of updating it so that it respects `recursive:` and then making `RecursiveDelete` a private helper method?

Reply:
Agree. Have renamed `RecursiveDelete` (because, most of its code can be used for non-recursive delete as well) to `DeleteDirectory`. It has an additional parameter `recursive` which is set to `true` by default. `DeleteDirectory` can now do both shallow and recursive deletes.